feature(discover) - Adding an order key to top events response

wmak opened the pull request:
- This will allow the frontend to order the tooltip, so that it matches
  what's in the table
- This modifies the tests so the test events have an order from count
  too

wmak committed:
feature(discover) - Adding an order key to top events response

- This will allow the frontend to order the tooltip, so that it matches
  what's in the table
- This modifies the tests so the test events have an order from count
  too

wmak commented on src/sentry/api/serializers/snuba.py:
```diff
@@ -314,5 +314,7 @@ def serialize(self, result, column="count"):
 
         if result.data.get("totals"):
             res["totals"] = {"count": result.data["totals"][column]}
+        if "order" in result.data:
```
can't use `result.data.get("order")` since order starts at `0` and is falsey

wmak replied:
Hmm, yeah I could add that too. in a multiaxis topevent query would we want the result to be like this?:
```
{
    "result_key": {
        "order": 0,
        "axis1": {
              "order": 0,
              "data": []...
```

markstory replied:
If not I can default to 0 in the front-end.

wmak replied:
This is a good point though, right now multi axis top events results would look like:
```
{
    "result_key": {
        "axis1": {
              "order": 0,
              "data": []...
```
so I'll move order up a level first to match below before merging :+1:
```
{
    "result_key": {
        "order": 0,
        "axis1": {
              "data": []...
```

markstory replied:
Right now we get

```json
{
  "rpm()": {"data": [....]},
  "apdex()": {"data": [....]},
}
```
Instead if we got

```
{
  "rpm()": {"order": 0, "data": [....]},
  "apdex()": {"order": 1, "data": [....]},
}
```

That would make top5 and multi-y-axis requests shaped the same.

wmak replied:
Misunderstood your original comment, can definitely shape both responses the same. But what happens in the case of multi axis + top events?

wmak commented on src/sentry/snuba/discover.py:
```diff
@@ -897,17 +907,20 @@ def top_events_timeseries(
+        result_key = create_result_key(row, translated_groupby, issues)
+        results.setdefault(result_key, {"data": []})["data"].append(row)
+    # Using the top events add the order to the results
+    for index, item in enumerate(top_events["data"]):
```
result["data"] won't have the orderby that `top_events` has. So gotta iterate twice. top_events should only be at most `limit` (5)

markstory reviewed (approved):
Looks good. I'll be able to get the results sorted consistently in the front-end with this.

wmak committed:
feat: Order for multi yaxis is order of columns

- Also moves order of top events so its the same between multi yaxis and
  single axis
